### PR TITLE
bot-sharepoint-file-viewer nodejs sample

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**
Bot which shares SharePoint file which can be viewed in Teams file viewer

### CODE REVIEW

File: samples/bot-sharepoint-file-viewer/nodejs/.env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+MicrosoftAppId=
+MicrosoftAppPassword=
+ConnectionName=
+SharepointTenantName=
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/.env:4]: Rename it to SharePointTenantName

File: samples/bot-sharepoint-file-viewer/nodejs/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+Using this Nodejs sample, a bot can upload file to SharePoint site and same files can be viewed in Teams file viewer.
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/README.md:1]: a bot with capability to upload files to SharePoint site and same files can be viewed in Teams file viewer

File: README.md [hunk 3]
```diff
@@ -66,6 +66,7 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 |9| Universal bots  | Teams catering bot demonstrating how to use Universal bots in Teams | [View](samples/bot-teams-catering/csharp) |  | |
 |10| Sequential workflow adaptive cards | Demonstrating on how to implement sequential flow, user specific view and upto date adaptive cards in bot. | [View][sequential#workflow#csharp] | [View][sequential#workflow#js]
 |11| Channel messages with RSC permissions | Demonstrating on how a bot can receive all channel messages with RSC without @mention. | [View][messageswithrsc#csharp] |[View][messageswithrsc#js]
+|12| Bot with SharePoint file viewer | This sample demos a bot which uploads file to SharePoint site and same files can be viewed in Teams file viewer. ||[View][botfileviewer#js]
```
Comment (v-chetsh) [inline README.md:69]: This sample demos a bot with capability to upload files to SharePoint site and same files can be viewed in Teams file viewer.

File: samples/bot-sharepoint-file-viewer/nodejs/simpleGraphClient.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    async getContentList(siteId, driveId) {
+        return await this.graphClient
+            .api('/sites/'+siteId+'/drives/'+driveId+'/root/children').version('beta')
+            .get().then((res) => {
+                return res;
+            });
+    }
+
+    /**
+    * This endpoin will upload the user's file to SharePoint drive .
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/simpleGraphClient.js:60]: endpoint

Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/simpleGraphClient.js:60]: Remove extra space before dot.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed and provided my comments.

File: samples/bot-sharepoint-file-viewer/nodejs/appPackage/manifest.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "privacyUrl": "https://www.microsoft.com/privacy",
+    "termsOfUseUrl": "https://www.microsoft.com/termsofuse"
+  },
+  "name": {
+    "short": "Prithvi File explorer",
+    "full": "Fetch groupchat messages with graph api"
+  },
+  "description": {
+    "short": "Prithvi File explorer",
+    "full": "This sample demos a bot which fetch and archive groupchat messages"
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/appPackage/manifest.json:19]: Is this full description correct?

File: samples/bot-sharepoint-file-viewer/nodejs/ssoOauthHelpler.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        // Only one of these token exchange requests should be processe by the bot.  For a distributed bot in production,
+        // this requires a distributed storage to ensure only one token exchange is processed.
+
+        // This example utilizes Bot Framework IStorage's ETag implementation for token exchange activity deduplication.
+
+        // Create a StoreItem with Etag of the unique 'signin/tokenExchange' request
+        const storeItem = {
+            eTag: turnContext.activity.value.id
+        };
+
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/ssoOauthHelpler.js:37]: Remove extra empty lines

File: samples/bot-sharepoint-file-viewer/nodejs/appPackage/manifest.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.10/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.10",
```
Comment (v-chetsh) [inline samples/bot-sharepoint-file-viewer/nodejs/appPackage/manifest.json:3]: Why have you pushed **sharepoint.zip** file??

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Reviewed and provided my feedback.

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**